pipettes/firmware/motor_control.cpp

```diff
@@ -0,0 +1,95 @@
+#include "pipettes/firmware/motor_control.hpp"
+
+#include "FreeRTOS.h"
+#include "common/core/bit_utils.hpp"
+#include "firmware/common/motor.h"
+#include "stm32g4xx_hal.h"
+#include "task.h"
+
+using namespace motor_control;
+
+/*
+ * Motor Control and namespace.
+ * Private:
+ * build_command takes in a command address, a reference
+ * to the data to be transmitted and the output array to
+ * store this in.
+ * reset_data takes nothing and resets the global data variable
+ *
+ * Public Globals:
+ * data, the data returned from the driver
+ * status, the status of the driver
+ *
+ * Public:
+ * setup, sets the initial configurations of the driver
+ * set_speed, sets a global variable called speed
+ * get_speed, gets a global variable called speed
+ * move, moves the motor for an unlimited amount of time
+ * get_status, get the status of the GSTAT register
+ */
+
+/*
+ * Private Functions
+ */
+void MotorControl::build_command(uint8_t command, const uint32_t& command_data,
```

Comment:
are commands always 5 bytes? if they can ever be different sizes we should probably have this return the output iterator.

Reply:
If we stick with TMC2130, I think they are always going to be 5 bytes.

Reply:
Honestly if this is all going to be 5 byte commands we could just build the array in this function and return it on the stack:
```c++
auto MotorControl::build_command(uint8_t command, uint32_t command_data) -> BufferType {
   auto tx_buffer = BufferType{0};
   ...
   return tx_buffer;
}
```
It's normally a no-no in cpp to return objects by-value but especially if the object is small, whatever.

Could also get a little fancy with returning via move which would make the compiler possibly emit zero-copy code but that too is pointless. 5 bytes is very small